Shared/Supporting Files/Views/SampleLink.swift

```diff
@@ -0,0 +1,144 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+
+struct SampleLink: View {
+    /// The sample to display.
+    private let sample: Sample
+    
+    /// The sample row to show as the link's label.
+    @ViewBuilder private var label: SampleRow
+    
+    init(_ sample: Sample) {
+        self.sample = sample
+        label = SampleRow(
+            name: AttributedString(sample.name),
+            description: AttributedString(sample.description)
+        )
+    }
+    
+    var body: some View {
+        NavigationLink {
+            SampleDetailView(sample: sample)
+                .id(sample.name)
+        } label: {
+            label
+        }
+    }
+}
+
+private extension SampleLink {
+    struct SampleRow: View {
+        /// The name string of the sample with attributes.
+        let name: AttributedString
+        
+        /// The description string of the sample with attributes.
+        let description: AttributedString
+        
+        /// A Boolean value indicating whether the sample's description is showing.
+        @State private var isShowingDescription = false
+        
+        /// The names of the favorited samples loaded from user defaults.
+        @AppStorage(UserDefaults.favoritedSamplesKey) private var favoritedNames = [String]()
+        
+        /// A Boolean value indicating whether the sample is favorited.
+        private var isFavorited: Bool {
+            favoritedNames.contains(String(name.characters))
+        }
+        
+        var body: some View {
+            ZStack {
+                if isFavorited {
+                    HStack {
+                        Image(systemName: "star.fill")
+                            .resizable()
+                            .aspectRatio(contentMode: .fit)
+                            .frame(width: 8)
+                            .padding(.leading, -13)
+                        Spacer()
+                    }
+                }
```

Comment:
Note to the reviewers: I was trying to emulate Apple Music's favorite star here, but I am not super happy with this solution since it is not future-proof, and it gets cut off when the context menu is showing on iPad. The alternatives are to either have the star as part of the title (then the title is offset compared to all the other unfavorited samples), or to get rid of this completely. Let me know what you think.

Reply:
Given that we cannot have something like Apple music, here are a few other ideas

- A normal sized star placed like Files app's tags may be acceptable
![image](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/9660181/b052c82d-9c8b-4e22-9310-dedd506098db)
- If it makes the sample name display too short, we may use the info button in an innovative way. Toggle a background yellow color to indicate a sample is favorited? ☆ -> ⭐ 

I realized that the star UI also resembles the messages unread blue dot. I'll try to find if it is sth we can mimic within SwiftUI.